driver/glu_driver.hpp

```diff
@@ -50,20 +50,37 @@ T sigmoid(T x)
 }
 
 template <typename Tgpu, typename Tcheck>
-int mloGLUForwardContiguousRunHost(const Tgpu* input,
+int mloGLUForwardContiguousRunHost(miopenTensorDescriptor_t inputDesc,
+                                   const Tgpu* input,
                                    miopenTensorDescriptor_t outputDesc,
-                                   Tcheck* outputHost)
+                                   Tcheck* outputHost,
+                                   uint32_t dim)
 {
     auto output_numel    = miopen::deref(outputDesc).GetElementSize();
-    auto inputFirstHalf  = input;
-    auto inputSecondHalf = input + output_numel;
+    auto input_lengths   = miopen::deref(inputDesc).GetLengths();
+    size_t dim_size      = input_lengths[dim];
+    size_t half_dim_size = dim_size / 2;
+    size_t inner_size    = 1;
+
+    for(size_t i = dim + 1; i < input_lengths.size(); i++)
+    {
+        inner_size *= input_lengths[i];
+    }
 
     int ret = 0;
 
     for(size_t o = 0; o < output_numel; o++)
     {
-        Tcheck valA   = static_cast<Tcheck>(inputFirstHalf[o]);
-        Tcheck valB   = static_cast<Tcheck>(inputSecondHalf[o]);
+        size_t outer_idx = o / (dim_size * inner_size);
+        o                = o % (dim_size * inner_size);
```

Comment:
@cognaiger9 I dont get it, why you change `o` here? Could you please explain it?

Reply:
I misunderstood your review earlier, but I have reverted to older version and properly renamed the functions.